doc/lua_api.txt

```diff
@@ -217,18 +217,40 @@ registered callbacks.
 `minetest.settings` can be used to read custom or existing settings at load
 time, if necessary. (See [`Settings`])
 
-### `models`
+#### Media
 
-Models for entities or meshnodes.
+All game assets, such as textures, sounds and models, are considered "media".
+Media is identified by unique names and SHA-1 hashes and cached on the client.
 
-### `textures`, `sounds`, `media`
+The following file formats are supported:
 
-Media files (textures, sounds, whatever) that will be transferred to the
-client and will be available for use by the mod.
+* Images: `.png`, `.jpg`, `.bmp`, `.tga`, `.pcx`, `.ppm`, `.psd`, `.wal` and `.rgb`
+* Sound: `.ogg`
+* Models: `.x`, `.b3d`, `.md2`, and `.obj`
+* Translations: `.tr` (custom format), not transferred to the client
+* Shaders: `.glsl`
 
-### `locale`
+There are 5 corresponding media directories, each for certain types of media:
 
-Translation files for the clients. (See [Translations])
+* `textures`: for most visual aspects
+* `sounds`: for `minetest.sound_play`
+* `models`: for entities or meshnodes
+* `locale`: translation files for the client (See [Translations])
+* `shaders`: custom graphical effects
+
+There also is the `media` directory for all kinds of media. Putting files in the wrong folders works but is frowned upon.
+
+##### Shaders
+
+Custom shaders to be sent to the client. GLSL version currently is `120`. **Experimental.**
```

Comment:
```suggestion
Custom shaders to be sent to the client. GLSL version currently is `120`.

**Experimental:** this may receive breaking changes in the future, without warning.
```